modules/discriminators.py

```diff
@@ -74,7 +74,7 @@ async def generate_comparison_claude_legacy(
         }
     )
     generation = await hooks.generate(
-        messages=wrapped_messages,
+        messages=[OpenaiChatMessage(**msg) for msg in wrapped_messages],
```

Comment:
Hm, this makes me think that we shouldn't be defining the Message class in base.py, which is just a stronger-typed version of OpenaiChatMessage.